stdlib/public/runtime/DynamicCast.cpp

```diff
@@ -126,9 +126,9 @@ static HeapObject * getNonNullSrcObject(OpaqueValue *srcValue,
 
   std::string srcTypeName = nameForMetadata(srcType);
   std::string destTypeName = nameForMetadata(destType);
-  const char * const msg = "Found unexpected null pointer value"
-                    " while trying to cast value of type '%s' (%p)"
-                    " to '%s' (%p)%s\n";
+  const char * const msg = "Non-Optional value of type '%s' (%p)"
+                    " holds a null pointer?!"
```

Comment:
```suggestion
                    " holds a null pointer, which is not allowed by the type system."
```

Thinking it's good to spell it out even more, in case someone doesn't understand why it's bad to have null in a non-optional.

Reply:
Revised message further:
> Found a null pointer in a value of type '%s'.  Non-Optional values are not allowed to hold null pointers. (Detected while casting to '%s')